document distributed tracing with highlight (#6319)
highlight/highlight: docs-content/getting-started/2_frontend-backend-mapping.md

- [Python Backend Integration](4_backend-sdk/python)

## Distributed Tracing

> Your backend might be a distributed system with multiple services. Say, for example, your
> frontend Next.js application uses a Next.js backend which in turn makes HTTP requests to
> a Python FastAPI microservice. You would want errors and logs from your Python service to be
> attributes to the frontend sessions in Highlight.

Reviewer: ```suggestion
Your backend might be a distributed system with multiple services. Say, for example, a
frontend Next.js application with a Next.js backend ,which makes HTTP requests to
a Python FastAPI microservice. In a case like that, you may want errors and logs from your Python service to be
attributed to the frontend sessions in Highlight.
```

> Our frontend -> backend tracing is based on the `x-highlight-request` HTTP header. If your Next.js backend issues an HTTP request to a FastAPI backend and you forward the `x-highlight-request` header along, the trace will carry over information about the frontend session.

Reviewer: ```suggestion
Our frontend -> backend tracing uses the `x-highlight-request` HTTP header to attribute frontend requests with backend errors and logs. So in the case of the example above, assuming all of your services have the highlight sdk installed, if your Next.js backend peforms an HTTP request to a FastAPI backend and you forward the `x-highlight-request` header along, the trace will carry over information about the frontend session.
```

> A more complex application might not make HTTP requests between backend services, but instead
> use a message broker like Kafka to queue up jobs. In that case, you'll need to add a way to
> store the `x-highlight-request` you receive from the frontend along with your enqueued messages.
> The service that consumes the messages can pass the value to the highlight SDK via custom
> error wrapping or logging code.

Reviewer: ```suggestion
A more complex application might not make HTTP requests between backend services, however. Instead it may
use a message broker like Kafka to queue up jobs. In that case, you'll need to add a way to
store the `x-highlight-request` you receive from the frontend along with your enqueued messages.
The service that consumes the messages can then pass the value to the highlight SDK via custom
error wrapping or logging code as per usual.
```
